src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -168,6 +169,8 @@ public void syncPostBatch(final String topicId, final List<BatchItem> batch) thr
         final CountDownLatch done = new CountDownLatch(batch.size());
 
         for (final BatchItem item : batch) {
+            Preconditions.checkNotNull(item.getPartition(),
```

Comment:
Well, I just wanted to be null-safe.
The partitioning phase can be changed in future and the null can appear here.